### PR TITLE
feature: add supernode piece downloaded size metrics

### PR DESCRIPTION
**Pull request by yeya24**
Signed-off-by: yeya24 <yb532204897@gmail.com>

### Ⅰ. Describe what this PR did
Add a counter type metric to count total size of pieces downloaded from supernode by dfget.

### Ⅱ. Does this pull request fix one issue?

NONE

### CODE REVIEW

File: supernode/util/range_util.go [hunk 1]
```diff
@@ -26,6 +26,30 @@ const (
 	separator = "-"
 )
 
+// CalculatePieceSize calculates the size of piece
+// according to the parameter range.
+func CalculatePieceSize(rangeStr string) int64 {
```
Comment (allencloud) [inline supernode/util/range_util.go]: Could you add unit test case for this function? Thanks.
I think everyone should keep in mind that test case is the good way to guarantee the software quality. ❤️

Comment (yeya24) [reply supernode/util/range_util.go]: Thanks for pointing out this. Updated!

**Comment by yeya24** [comment]
Any progress for this PR?

File: supernode/util/range_util.go (hunk 1 above)
Comment (Starnop) [reply supernode/util/range_util.go]: I think you can compress `CalculatePieceSize` and `CalculatePieceNum` methods.

File: docs/user_guide/metrics.md [hunk 2]
```diff
@@ -18,6 +18,7 @@ This doc contains all the metrics that Dragonfly components currently support. N
 - dragonfly_supernode_schedule_duration_milliseconds{peer} - duration for task scheduling in milliseconds
 - dragonfly_supernode_trigger_cdn_total{} - total times of triggering cdn.
 - dragonfly_supernode_trigger_cdn_failed_total{} - total failed times of triggering cdn.
+- dragonfly_supernode_pieces_downloaded_size_bytes{} - total size of pieces downloaded from supernode in bytes. counter type.
```
[the remark this answers is not in the document]
Comment (yeya24) [reply docs/user_guide/metrics.md]: I think maybe make this metrics docs as a table is better, we can add a column called `metrics type` to represent this.

File: supernode/util/range_util.go (hunk 1 above)
Comment (yeya24) [reply supernode/util/range_util.go]: What do you mean by `compress` this two methods? Do you mean use one single method to return this two values?

**Comment by Starnop** [comment]
LGTM.

**Pull request merged**